spec/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source_spec.rb

```diff
@@ -0,0 +1,518 @@
+          result = described_class.last
+
+          expect(result).to be_an(described_class)
+          expect(result.scm_type).to eq("git")
+          expect(result.scm_branch).to eq("other_branch")
+          expect(result.status).to eq("error")
+          expect(result.last_updated_on).to be_an(Time)
+          expect(result.last_update_error).to start_with("Rugged::NetworkError")
+        end
+
+        it "clears last_update_error on re-sync" do
+          result = described_class.last
+
+          expect(result.status).to eq("error")
+          expect(result.last_updated_on).to be_an(Time)
+          expect(result.last_update_error).to start_with("Rugged::NetworkError")
+
+          expect(result.git_repository).to receive(:update_repo).and_call_original
+
+          result.sync
+
+          expect(result.status).to(eq("successful"))
+          expect(result.last_update_error).to(be_nil)
+        end
+      end
+    end
+
+    describe "#update_in_provider_queue" do
+      it "creates a task and queue item" do
+        record    = build_record
+        task_id   = record.update_in_provider_queue({})
+        task_name = "Updating #{described_class::FRIENDLY_NAME} (name=#{record.name})"
+
+        expect(MiqTask.find(task_id)).to(have_attributes(:name => task_name))
+        expect(MiqQueue.first).to(
+          have_attributes(
+            :instance_id => record.id,
+            :args        => [{:task_id => task_id}],
+            :class_name  => described_class.name,
+            :method_name => "update_in_provider",
+            :priority    => MiqQueue::HIGH_PRIORITY,
+            # :role        => "embedded_terraform",
+            :zone        => nil
+          )
+        )
+      end
+    end
+
+    describe "#delete_in_provider" do
+      it "deletes the record and removes the git dir" do
+        record = build_record
+        git_repo_dir = repo_dir.join(record.git_repository.id.to_s)
+
+        # expect(Notification).to(receive(:create!).with(notification_args('deletion', {})))
+        record.delete_in_provider
+
+        # Run most recent queue item (`GitRepository#broadcast_repo_dir_delete`)
+        MiqQueue.get.deliver
+
+        expect(record).to(be_deleted)
+
+        expect(git_repo_dir).to_not(exist)
+      end
+    end
+
+    describe "#delete_in_provider_queue" do
+      it "creates a task and queue item" do
+        record    = build_record
+        task_id   = record.delete_in_provider_queue
+        task_name = "Deleting #{described_class::FRIENDLY_NAME} (name=#{record.name})"
+
+        expect(MiqTask.find(task_id)).to(have_attributes(:name => task_name))
+        expect(MiqQueue.first).to(
+          have_attributes(
+            :instance_id => record.id,
+            :args        => [],
+            :class_name  => described_class.name,
+            :method_name => "delete_in_provider",
+            :priority    => MiqQueue::HIGH_PRIORITY,
+            # :role        => "embedded_terraform",
```

Comment:
Can this be uncommented?  Should we be testing that we're setting the right role?